src/DGtal/kernel/UnorderedSetByBlock.h

```diff
@@ -0,0 +1,1218 @@
+  /// occupancy is divided by four, the structure is four times faster
+  /// for traversal and approximately same speed for
+  /// queries/insertion/erase.
+  ///
+  /// Almost all standard operations of unordered_set in c++11 are implemented.
+  ///
+  /// @tparam Key the type of integral array.
+  /// @tparam TSplitter the type for splitting a key into a block and a bit (see \ref Splitter).
+  /// @tparam Hash the type that provides a hasher for Key.
+  /// @tparam KeyEqual the type that provides an equality comparator for Key.
+  /// @tparam UnorderedMapAllocator the type that provides an allocator for the underlying unordered_map container.
+  ///
+  /// Specialized versions of Splitter are written for usual digital
+  /// points of Z2i or Z3i.
+  ///
+  /// @code
+  /// #include "DGtal/kernel/UnorderedSetByBlock.h"
+  /// ...
+  /// typedef DGtal::PointVector< 3, int >  Point3i; // digital point in Z3;
+  /// std::unordered_set< Point3i >         aSet;    // usual data structure
+  /// DGtal::UnorderedSetByBlock< Point3i > aSet2;   // this data structure
+  /// // same code after for aSet or aSet2.
+  /// @endcode
+  template < typename Key,
+	     typename TSplitter = Splitter< Key >,
+	     class Hash = std::hash<Key>,
+	     class KeyEqual = std::equal_to<Key>,
+	     class UnorderedMapAllocator = std::allocator<
+	       std::pair<const Key, typename Splitter< Key >::Word >
+	       >
+	     >
+  struct UnorderedSetByBlock {
+    typedef UnorderedSetByBlock< Key, TSplitter, Hash, KeyEqual > Self;
+    typedef TSplitter               Splitter;
+    typedef typename Splitter::Word Word;
+    /// The underlying container, an unordered_map.
+    typedef std::unordered_map< Key, Word, Hash, KeyEqual,
+				UnorderedMapAllocator > Container;
+    
+    // Standard types
+    /// Key
+    typedef Key key_type;
+    /// Key
+    typedef Key value_type;
+    /// Unsigned integer type (usually std::size_t)
+    typedef typename Container::size_type size_type;
+    /// Signed integer type (usually std::ptrdiff_t)
+    typedef typename Container::difference_type difference_type;
+    /// Hash
+    typedef Hash     hasher;
+    /// KeyEqual
+    typedef KeyEqual key_equal;
+    /// Allocator
+    typedef typename Container::allocator_type allocator_type;
+    /// Reference to value_type/Key
+    typedef Key& reference;
+    /// Const reference to value_type/Key
+    typedef const Key& const_reference;
+    /// Pointer to value_type/Key
+    typedef Key* pointer;
+    /// Const Pointer to value_type/Key
+    typedef const Key* const_pointer;
+
+  public:
+    // ---------------------- iterators --------------------------------
+    
+    /// Read iterator on set elements. Model of ForwardIterator.
+    struct const_iterator
+      : public boost::iterator_facade< const_iterator, Key const,
+				       boost::forward_traversal_tag,
+				       Key const >
+    {
+      friend class UnorderedSetByBlock< Key, TSplitter, Hash, KeyEqual >;
```

Comment:
```suggestion
      friend struct UnorderedSetByBlock< Key, TSplitter, Hash, KeyEqual >;
```
or define `UnorderedSetByBlock` as a `class`.

Reply:
done